internal/cmd/gocdk/main.go

```diff
@@ -111,6 +111,11 @@ func osProcessContext() (*processContext, error) {
 	}, nil
 }
 
+// overrideEnv returns a copy of pctx.env that has vars appended to the end.
+func (pctx *processContext) overrideEnv(vars ...string) []string {
+	return append(pctx.env[:len(pctx.env):len(pctx.env)], vars...)
```

Comment:
My intent with this line is to always create a new slice, since I don't want to mutate `pctx.env` or its backing array. Thus, this makes `overrideEnv` safe to call from multiple goroutines. By setting the capacity to the length, `append` will always allocate unless `len(vars) == 0`.

Reply:
That is super subtle. I would suggest making it more explicit (e.g., just copy the source slice explicitly, then append. Or at the very least add a comment.

Reply:
Comment added.